### PR TITLE
[scalert] allow filtering on streamID for pick script

### PR DESCRIPTION
**Pull request by luca-s**
This PR adds a configuration option that allows to filter the picks for which the pick script is called based on the pick streamID.

### CODE REVIEW

File: apps/python/descriptions/scalert.xml [hunk 1]
```diff
@@ -85,6 +85,17 @@
 					received pick has one of the value(s).
 					</description>
 				</parameter>
+				<parameter name="phaseStreams" type="list::string" default="">
+					<description>
+					Start the pick script only when the stream (NET.STA.LOC.CHA)
+					of the received	pick belongs to the list. If empty, all picks
```
Comment (gempa-dirk) [inline apps/python/descriptions/scalert.xml:91]: Remove double spaces

File: apps/python/scalert.py [hunk 2]
```diff
@@ -123,6 +125,23 @@ def init(self):
...
             pass
 
+        self._phaseStreams = []
+        try:
+            phaseStreams = self.configGetStrings("constraints.phaseStreams")
+            for item in phaseStreams:
+                rule = item.strip()
+                # rule is NET.STA.LOC.CHA and the special charactes ? * | ( ) are allowed
+                if not re.fullmatch(r'[A-Z|a-z|0-9|\?|\*|\||\(|\)|\.]+', rule):
+                    seiscomp.logging.error("Wrong wafeformid in `constraints.phaseStreams`: %s" % item)
```
Comment (gempa-dirk) [inline apps/python/scalert.py:135]: wafeformid -> stream ID (not wafe!)

File: apps/python/scalert.py [hunk 3]
```diff
@@ -286,6 +305,11 @@ def init(self):
         else:
             seiscomp.logging.info(" + phase hints: no filter is applied")
 
+        if " ".join(self._phaseStreams):
+            seiscomp.logging.info(" + phase waveformid filter for picks: '%s'" % (" ".join(self._phaseStreams)))
```
Comment (gempa-dirk) [inline apps/python/scalert.py:309]: waveformid -> waveform ID, write capitals for abbreviations and separate them.

Comment (gempa-dirk) [reply apps/python/scalert.py:309]: stream ID instead of waveformid

File: apps/python/scalert.py [hunk 4]
```diff
@@ -419,6 +443,21 @@ def addObject(self, parentID, object):
...
+                            obj.waveformID().networkCode(), obj.waveformID().stationCode(), 
+                            obj.waveformID().locationCode(), obj.waveformID().channelCode())
+                    matched = False
+                    for rule in self._phaseStreams:
+                        if re.fullmatch(rule, waveformID):
+                            matched = True
+                            break
+                    if not matched:
+                        seiscomp.logging.debug(
+                                " + waveformID %s does not match constraints.phaseStreams rules"
```
Comment (gempa-dirk) [inline apps/python/scalert.py:457]: stream ID

**Review by gempa-dirk — CHANGES_REQUESTED** [review]

**Comment by luca-s** [comment]
Thanks for reviewing @gempa-dirk. I fixed the issues

**Comment by andres-h** [comment]
@cla-bot check

**Comment by andres-h** [comment]
@cla-bot check

**Comment by luca-s** [comment]
This should now be ready unless more changes are required from your side.

**Comment by gempa-jabe** [comment]
Please rebase your branch.

**Pull request merged**